test/diff.jl

```diff
@@ -33,6 +33,35 @@ dcsch = D(csch(t))
 @test isequal(expand_derivatives(D2(t)), 0)
 @test isequal(expand_derivatives(D2(5)), 0)
 
+# Vector calculus operators
+@variables vv(t,x,y)
+∇z = Gradient(z)
+∇xy = Gradient(x,y)
```

Comment:
People using this package are fond of using a large number of variables, so I'd recommend this just take a vector of variables.

I'm also wondering if it would be possible to have gradient just be `Differential([x,y])` -- use multiple dispatch to make `Differential{Vector}` behave like a gradient. But not very important

@YingboMa thoughts?

Reply:
When we do have vector-like variables, we can store the `symtype(x)` as parameter rather than `typeof(x)` and that would still continue to work given all the methods it touches are defined.

Reply:
>I'm also wondering if it would be possible to have gradient just be Differential([x,y]) -- use multiple dispatch to make Differential{Vector} behave like a gradient. But not very important

Instead of calling it the gradient, make it the total derivative, i.e. the Jacobian, which is then just the transpose of the gradient when you have a single output. I think that's what we actually want in this case.